File: Dart/lib/uncommonFromSentences.dart
class Solution {
  List<String> uncommonFromSentences(String s1, String s2) {
    var map = <String, int>{};
    for (var word in s1.split(' ')) {
      map[word] = (map[word] ?? 0) + 1;
    }
    for (var word in s2.split(' ')) {
      map[word] = (map[word] ?? 0) + 1;
    }
    return map.entries.where((e) => e.value == 1).map((e) => e.key).toList();
  }
}

File: Cs/Solution/UncommonFromSentences.cs
using System;
using System.Collections.Generic;

namespace Solution
{
	public partial class Solution
	{
		public string[] UncommonFromSentences(string s1, string s2)
		{
			string[] words1 = s1.Split(' ');
			string[] words2 = s2.Split(' ');
			Dictionary<string, int> freq = new();
			foreach (string word in words1)
			{
				freq[word] = freq.GetValueOrDefault(word, 0) + 1;
			}
			foreach (string word in words2)
			{
				freq[word] = freq.GetValueOrDefault(word, 0) + 1;
			}
			List<string> ret = new();
			foreach (KeyValuePair<string, int> kvp in freq)
			{
				if (kvp.Value == 1)
				{
					ret.Add(kvp.Key);
				}
			}
			return ret.ToArray();
		}
	}
}

